File: matrix-multiplication/matrix_multiplication.php
<?php
function matrixMultiply($a, $b) {
    $rowsA = count($a);
    $colsA = count($a[0]);
    $colsB = count($b[0]);
    
    $result = array_fill(0, $rowsA, array_fill(0, $colsB, 0));
    
    for ($i = 0; $i < $rowsA; $i++) {
        for ($j = 0; $j < $colsB; $j++) {
            for ($k = 0; $k < $colsA; $k++) {
                $result[$i][$j] += $a[$i][$k] * $b[$k][$j];
            }
        }
    }
    
    return $result;
}

$a = [[1, 2], [3, 4]];
$b = [[5, 6], [7, 8]];

$result = matrixMultiply($a, $b);
foreach ($result as $row) {
    echo implode(" ", $row) . "\n";
}
?>

File: matrix-multiplication/MatrixMultiplication.kt
fun matrixMultiply(a: Array<IntArray>, b: Array<IntArray>): Array<IntArray> {
    val rowsA = a.size
    val colsA = a[0].size
    val colsB = b[0].size
    
    val result = Array(rowsA) { IntArray(colsB) }
    
    for (i in 0 until rowsA) {
        for (j in 0 until colsB) {
            for (k in 0 until colsA) {
                result[i][j] += a[i][k] * b[k][j]
            }
        }
    }
    
    return result
}

fun main() {
    val a = arrayOf(intArrayOf(1, 2), intArrayOf(3, 4))
    val b = arrayOf(intArrayOf(5, 6), intArrayOf(7, 8))
    
    val result = matrixMultiply(a, b)
    result.forEach { row ->
        println(row.joinToString(" "))
    }
}
